packages/blade/src/components/Input/BaseInput/baseInputStyles.ts

```diff
@@ -78,29 +81,35 @@ export const getInputBackgroundAndBorderStyles = ({
   | 'isHovered'
   | 'isTextArea'
   | 'isDropdownTrigger'
+  | 'hasBorder'
 >): CSSObject => {
   // normal state
-  let backgroundColor = getIn(theme.colors, baseInputBackgroundColor.default);
-  let borderColor = getIn(theme.colors, baseInputBorderColor.default);
+  const backgroundColorTokens = hasBorder
+    ? baseInputBackgroundColor
+    : baseInputBorderlessBackgroundColor;
+  let backgroundColor = getIn(theme.colors, backgroundColorTokens.default);
+  let borderColor = hasBorder ? getIn(theme.colors, baseInputBorderColor.default) : 'transparent';
   let borderWidth = getIn(theme.border.width, baseInputBorderWidth.default);
 
   const baseInputState = getBaseInputState({ isFocused, isHovered, isDisabled });
 
-  backgroundColor = getIn(theme.colors, baseInputBackgroundColor[baseInputState]);
-  borderColor = getIn(theme.colors, baseInputBorderColor[baseInputState]);
+  backgroundColor = getIn(theme.colors, backgroundColorTokens[baseInputState]);
+  borderColor =
+    !hasBorder && baseInputState !== 'focused'
+      ? 'transparent'
+      : getIn(theme.colors, baseInputBorderColor[baseInputState]);
```

Comment:
```suggestion
    hasBorder || baseInputState === 'focused'
      ? getIn(theme.colors, baseInputBorderColor[baseInputState])
      : theme.colors.transparent;
```

Is this the same thing 🤔

Reply:
Updated